### PR TITLE
Update docs to clarify supported Windows and PowerShell versions

### PR DESCRIPTION
**Pull request by pauby**
## Description Of Changes
Updated the documentation to  include information on the supported Windows client and server operating systems, and the supported PowerShell versions.

## Motivation and Context
Our documentation contained outdated and misleading infromation on what we supported.

## Testing

* [x] I have previewed these changes using the [Docker Container](https://github.com/chocolatey/docs/tree/master/.devcontainer) or another method before submitting this pull request.

## Change Types Made

* [ ] Minor documentation fix (typos etc.).
* [x] Major documentation change (refactoring, reformatting or adding documentation to existing page).
* [ ] New documentation page added.
* [ ] The change I have made should have a video added, and I have raised an issue for this.
    * Issue #

## Change Checklist

* [ ] Requires a change to menu structure (top or left-hand side)/
* [ ] Menu structure has been updated

## Related Issue
N/A

### CODE REVIEW

File: input/en-us/chocolatey-components-dependencies-and-support-lifecycle.md [hunk 1]
```diff
@@ -33,23 +33,48 @@ Once a product reaches the second major release, the following will apply:
...
+
+[11homepro]: https://learn.microsoft.com/en-us/lifecycle/products/windows-11-home-and-pro
+[11entedu]: https://learn.microsoft.com/en-us/lifecycle/products/windows-11-enterprise-and-education
+[10homepro]: https://learn.microsoft.com/en-us/lifecycle/products/windows-10-home-and-pro
+[10entedu]: https://learn.microsoft.com/en-us/lifecycle/products/windows-10-enterprise-and-education
+[serv2022]: https://learn.microsoft.com/en-us/lifecycle/products/windows-server-2022
+[serv2019]: https://learn.microsoft.com/en-us/lifecycle/products/windows-server-2019
+[serv2016]: https://learn.microsoft.com/en-us/lifecycle/products/windows-server-2016
+[serv2012r2]: https://learn.microsoft.com/en-us/lifecycle/products/windows-server-2012-r2
+[serv2012]: https://learn.microsoft.com/en-us/lifecycle/products/windows-server-2012
```
Comment (pauby) [inline input/en-us/chocolatey-components-dependencies-and-support-lifecycle.md:80]: I _really_ dislike this. It was the only way for the table to actually still be maintainable. The code without it was a mess.

Sorry.

**Review by gep13 — APPROVED** [review]
LGTM!

File: input/en-us/chocolatey-components-dependencies-and-support-lifecycle.md (hunk 1 above)
Comment (gep13) [reply input/en-us/chocolatey-components-dependencies-and-support-lifecycle.md]: This change makes sense to me!

**Comment by gep13** [comment]
@pauby thanks for getting this added.  I have enabled auto-merge on this PR.

**Pull request merged**